### PR TITLE
Swo 102 support playlist

### PR DESCRIPTION
**Pull request by shinaBR2**
## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 📝 Documentation update
- [ ] 🧹 Code refactor
- [ ] ⚡️ Performance improvements
- [ ] ✅ Test updates

## How Has This Been Tested?

- [ ] Added/updated unit tests
- [ ] Manual testing

## Checklist

- [ ] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have updated the version number (if applicable)

## Summary by CodeRabbit

- **New Features**
	- Introduced more descriptive, intuitive routes for accessing video and playlist details, offering smoother navigation and lazy-loaded content.
	- Added a new utility function for generating link properties based on media types.
	- Implemented new GraphQL fragments for user, video, and playlist details to streamline data fetching.
	- Added a new hook for loading playlist details with enhanced data transformation.

- **Refactor**
	- Enhanced authentication and media loading processes to improve performance and reliability across video detail, history, and home screens.
	- Optimized data fetching and transformation for robust display of video content.
	- Updated component props structures to improve clarity and functionality.
	- Streamlined route generation logic for better modularity and maintainability.

- **Tests**
	- Expanded test coverage for navigation, video transformations, and data handling to ensure a consistently smooth and stable viewing experience.
	- Introduced new tests for transformation functions and route generation logic.
	- Added comprehensive tests for the `useLoadPlaylistDetail` hook and transformation functions, validating various edge cases.
	- Enhanced tests for the `RelatedList` component to verify link properties based on playlist presence.
	- Added new tests for the `VideoCard` component to validate rendering behavior based on media types.

### CODE REVIEW

File: packages/core/src/watch/query-hooks/transformers.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    thumbnailUrl: video.thumbnailUrl || '',
+    source: video.source,
+    slug: video.slug,
+    duration: video.duration || 0,
+    createdAt: video.createdAt,
+    user,
+    lastWatchedAt: history?.last_watched_at ?? null,
+    progressSeconds: history?.progress_seconds ?? 0,
+  };
+};
```
[the remark this answers is not in the document]
Comment (shinaBR2) [reply packages/core/src/watch/query-hooks/transformers.ts:33]: user is always there because I have validation in database, so it's fine here.

File: packages/ui/src/watch/home-page/container/index.test.tsx [hunk 2]
```diff
@@ -12,6 +12,16 @@ vi.mock('../../videos/video-card', () => ({
...
 
+vi.mock('../utils', () => ({
+  genlinkProps: (video: { id: string; slug: string }) => ({
+    to: '/video/$slug-$id',
+    params: {
+      slug: video.slug,
+      id: video.id,
+    },
+  }),
+}));
```
[the remark this answers is not in the document]
Comment (shinaBR2) [reply packages/ui/src/watch/home-page/container/index.test.tsx:23]: I thought it should be test by `genLinkProps` method itself?

File: packages/ui/src/watch/videos/video-card/index.test.tsx [hunk 3]
```diff
@@ -20,9 +29,10 @@ const renderWithAct = async (component: React.ReactElement) => {
   return result!;
 };
 
-describe('VideoCard Component', () => {
+describe.only('VideoCard Component', () => {
```
[the remark this answers is not in the document]
Comment (shinaBR2) [reply packages/ui/src/watch/videos/video-card/index.test.tsx]: I fixed it.

File: packages/core/src/watch/query-hooks/videos/index.ts [hunk 4]
```diff
@@ -29,47 +21,16 @@ interface LoadVideosProps {
...
-    source: video.source,
-    slug: video.slug,
-    duration: video.duration || 0,
-    createdAt: video.createdAt,
-    user,
-    lastWatchedAt: history?.last_watched_at ?? null,
-    progressSeconds: history?.progress_seconds ?? 0,
-  };
+  return sorted;
 };
```
[the remark this answers is not in the document]
Comment (shinaBR2) [reply packages/core/src/watch/query-hooks/videos/index.ts]: I updated it.

File: packages/ui/src/watch/videos/video-card/index.tsx [hunk 5]
```diff
@@ -37,85 +35,90 @@
...
+    );
+  }
+
+  return null;
 };
 
-const VideoCard = ({ video, asLink, LinkComponent }: VideoCardProps) => {
+const VideoCard = (props: VideoCardProps) => {
+  const { video, asLink, LinkComponent, linkProps } = props;
+
```
[the remark this answers is not in the document]
Comment (shinaBR2) [reply packages/ui/src/watch/videos/video-card/index.tsx:116]: It's already there in the `packages/ui/src/watch/videos/types.ts`
